homeassistant/components/onvif/camera.py

```diff
@@ -105,7 +124,9 @@
                 camera for camera in all_cameras if camera.entity_id in entity_ids
             ]
         for camera in target_cameras:
-            await camera.async_perform_ptz(pan, tilt, zoom)
+            await camera.async_perform_ptz(
+                pan, tilt, zoom, distance, speed, move_mode, continuous_duration
+            )
 
     hass.services.async_register(
         DOMAIN, SERVICE_PTZ, async_handle_ptz, schema=SERVICE_PTZ_SCHEMA
```

Comment:
Please register the custom service under the onvif domain instead of the camera domain. Custom services must be registered under the integration domain.

Reply:
To adress your comment, do i need to remove this import:
`from homeassistant.components.camera.const import DOMAIN`
and replace it after imports by:
`DOMAIN="onvif"`

Is it correct ?

Reply:
I've tested it that way and it did the trick. 
Just ignore my last comment if your ok with that way to do.